// Camera/cameraApi.js
var PythonShell = require('python-shell');
var fs = require('fs');

var image_data = './image_data.json';
var camera_data = './camera.json';

var options = {
    mode: 'text',
    pythonOptions: ['-u'],
    scriptPath: '/home/pi/pywork',
    args: []
};


module.exports = {
    pics: function pics(cb) {
        console.log("Start takin pics")
        var shell = new PythonShell('click_shoot_run.py', options);
        shell.on('message', function (message) {
            json_file = message
            fs.writeFile(image_data, json_file, function (err) {
                if (err) {
                    console.log(err);
                } else {
                    console.log("JSON saved to " + image_data);
                    cb()
                }
            });

        });
        shell.end(function (err) {
            if (err) {
                throw err;
            }
        });
    },

    one: function one(flash, cb) {
        var shell = new PythonShell('one_pic.py', {
            mode: 'text',
            pythonOptions: ['-u'],
            scriptPath: '/home/pi/pywork',
            args: [flash]
        });
        shell.on('message', function (message) {
            cb(message)
        })
        shell.end(function (err) {
            if (err) {
                cb('/images/err.png');
            }
        });
    },

    pico: function pico(cb) {
        impath = '/images/1.jpg'
        setTimeout(siko, 3500)
        function siko() {
            cb(impath)
        }
    },

    get_data: function get_data(cb) {
        var shell = new PythonShell('json_gen.py', options);
        shell.on('message', function (message) {
            json_file = message
            fs.writeFile(camera_data, json_file, function (err) {
                if (err) {
                    console.log(err);
                } else {
                    console.log("JSON saved to " + camera_data);
                }
            });
        });
        shell.end(function (err) {
            if (err) {
                var state = 0
                cb(state, 0);
            } else {
                var space = require('../../camera.json')
                space = space.freeimg;
                var state = 1
                cb(state, space)
            }
        });

    },

    get_data_off: function get_data_off(cb) {
        //function kirk(){
            var state = 1
            cb(state)
        //}
        //setTimeout(kirk,1000)
    }


}
