// Camera/cameraApi.js
var PythonShell = require('python-shell');
var fs = require('fs');

var outputFilename = './image_data.json';

var options = {
    mode: 'text',
    pythonOptions: ['-u'],
    scriptPath: '/home/pi/pywork',
    args: []
};

module.exports={
    pics : function pics() {
        var shell = new PythonShell('shooting_test.py', options);
        shell.on('message', function (message) {
            json_file=message
            fs.writeFile(outputFilename, json_file, function(err) {
                if(err) {
                    console.log(err);
                } else {
                    console.log("JSON saved to " + outputFilename);
                }
            });

        });
    },

    one : function one(){
        var target
        var shell = new PythonShell('one_pic.py', options);
        shell.on('message', function(message, err){
           target =message
            if(err) {
                target =0
                console.log(err);
            } else {
                console.log("succes");
            }
        })
        return target;
    },


    pico : function pico(){
        return 1;
    }
}
